bindings/java/src/org/sleuthkit/datamodel/TimelineManager.java

```diff
@@ -519,6 +511,44 @@ private long addEventDescription(long dataSourceObjId, long fileObjId, Long arti
 			caseDB.releaseSingleUserCaseWriteLock();
 		}
 	}
+	
+	/**
+	 * Returns an event description id for an existing event.
+	 * 
+	 * @param dataSourceObjId	Existing data source object id
+	 * @param fileObjId			Existing content object id
+	 * @param artifactID		Existing artifact id
+	 * @param fullDescription	Full event description
+	 * @param connection		Database connection
+	 * 
+	 * @return The id of an existing description or null if none what found.
+	 * 
+	 * @throws TskCoreException 
+	 */
+	private Long getEventDescription(long dataSourceObjId, long fileObjId, Long artifactID,
+			String fullDescription, CaseDbConnection connection) throws TskCoreException {
+
+		String query = "SELECT event_description_id FROM tsk_event_descriptions "
+				+ "WHERE data_source_obj_id = " + dataSourceObjId
+				+ " AND content_obj_id = " + fileObjId
+				+ " AND artifact_id = " + artifactID
+				+ " AND full_description " + (fullDescription != null ? "= '" + fullDescription + "'" : "IS null");
+
+		caseDB.acquireSingleUserCaseReadLock();
+		try (ResultSet resultSet = connection.createStatement().executeQuery(query)) {
```

Comment:
It might be safer to use a PreparedStatement here.

Reply:
The PreparedStatements do not buy us anything unless we are saving them for future use.  Also I tried it with a prepared statement and it didn't work.  This worked.

Reply:
I'm fairly certain a string like "Richard's Phone Call" for a description will cause a failure because of the single quote.

Reply:
While I don't disagree with your point. We control the code the code that creates the description therefore I don't understand how the quotes appeared in the first place.  We have many places in the code with select statements similar to mine that use single quotes.